### DISCUSSION TITLE
Add manual method?

### DISCUSSION THREAD

**Discussion by dlithio** [wkarney/minted-export#7] [General]
Any thoughts on adding a more manual method? I had trouble getting selenium working (kept getting `ImportError: cannot import name 'PhantomJS' from 'selenium.webdriver'`), so I ended up just following these steps below. The key parts are lifted from the code in this repo.

1. Open a browser and log into [minted](https://www.minted.com/).
2. Head to this URL - https://addressbook.minted.com/api/contacts/contacts/?format=json
3. If in Chrome, right click, "Save as...", then save it to `download.json`. Other browsers will likely be similar.
4. Run the following code in the same folder where `download.json` is located:
```python

import json
import pandas as pd
with open('download.json') as f:
    data = json.load(f)
address_book = pd.DataFrame(data)
address_book.to_excel("minted-addresses-api.xlsx")
address_book.to_csv("minted-addresses-api.csv", index=False)
```

If this sounds like an alright idea, I'd be happy to write a PR adding a section to the README and a python script contain the code in step 4.

**Comment by wkarney** [comment]
That sounds like a very good idea. Selenium is cool when it works but obviously (from people's comments) setting up the configuration on everyone's machines can create some headaches. The slightly more manual workaround is great; please go ahead fork the repo and send me a PR with your updates. 

My only suggestions are:
* Maybe we call that new script `convert_json.py` or something similar
* As you mentioned, make sure to add documentation in the [README](https://github.com/wkarney/minted-export/blob/master/README.md) walking through the steps as you have above.
* Add entries in the [gitignore](https://github.com/wkarney/minted-export/blob/master/.gitignore) to catch csv and xlsx files (in case people choose to run from within the project directory)
* Make sure to run install the pre-commit hook locally (`pre-commit install`) so the formatting is consistent and we get that nice pretty ✅ on the PR

**Comment by wkarney** [comment]
I added a walkthrough of this manual method and the converter script, see #10. Thanks for the suggestion!

**Comment by russhearn** [comment]
Hello, I tried the automated and manual method. 

Automated didn't work, manual did but I had to create the "data" directory. Before I created the directory I got this error about file not found:

(env) russ@RCH-MBPc minted-export % python convert_json.py
Traceback (most recent call last):
  File "convert_json.py", line 10, in <module>
    address_book.to_excel("./data/minted-addresses-converted.xlsx")
  File "/Users/russ/minted-export/env/lib/python3.8/site-packages/pandas/core/generic.py", line 2186, in to_excel
    formatter.write(
  File "/Users/russ/minted-export/env/lib/python3.8/site-packages/pandas/io/formats/excel.py", line 815, in write
    writer = ExcelWriter(  # type: ignore[abstract]
  File "/Users/russ/minted-export/env/lib/python3.8/site-packages/pandas/io/excel/_openpyxl.py", line 30, in __init__
    super().__init__(
  File "/Users/russ/minted-export/env/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 799, in __init__
    self.handles = get_handle(
  File "/Users/russ/minted-export/env/lib/python3.8/site-packages/pandas/io/common.py", line 648, in get_handle
    handle = open(handle, ioargs.mode)
FileNotFoundError: [Errno 2] No such file or directory: './data/minted-addresses-converted.xlsx'

... so I just made an empty "data" directory and it worked. 

Thank you!
